File: 백준/Bronze/16917. 양념 반 후라이드 반/양념 반 후라이드 반.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int a,b,c,x,y;
    cin >> a >> b >> c >> x >> y;
    int price = 0;
    
    if(a+b > c*2)
    {
        for(int i=0;i<min(x,y);i++) price = price + (c * 2);

        if(a > c*2 && x > y) price = price + c*2*(x-y);
        else if (b > c*2 && x < y) price = price + c*2*(y-x);
        else price = price + (x > y ? a : b)*(max(x,y) - min(x,y));
    }
    else  price = a*x + b*y;
    
    cout << price;
}

File: 백준/Bronze/16917. 양념 반 후라이드 반/양념 반 후라이드 반.java
import java.util.*;

public class Main {
    public static void main(String args[]) {
      Scanner s = new Scanner(System.in);
      int a = s.nextInt(),b = s.nextInt(),c = s.nextInt(), x = s.nextInt(), y = s.nextInt();
      int price = 0;
      if(a+b > c*2)
      {
          for(int i=0;i<Math.min(x,y);i++) price = price + (c * 2);

          if(a > c*2 && x > y) price = price + c*2*(x-y);
          else if (b > c*2 && x < y) price = price + c*2*(y-x);
          else price = price + (x > y ? a : b)*(Math.max(x,y) - Math.min(x,y));
      }
        else price = a*x + b*y;
      System.out.println(price);
    }
}